provider/fs/src/export/fs_exporter.rs

```diff
@@ -55,46 +57,15 @@ impl Default for ExporterOptions {
 pub struct FilesystemExporter {
     root: PathBuf,
     manifest: Manifest,
-    alias_collection: Option<AliasCollection<Vec<u8>>>,
-    serializer: Box<dyn AbstractSerializer>,
-}
-
-impl Drop for FilesystemExporter {
-    fn drop(&mut self) {
-        if self.alias_collection.is_some() {
-            panic!("Please call flush before dropping FilesystemExporter");
-        }
-    }
-}
-
-impl DataExporter<SerializeMarker> for FilesystemExporter {
-    fn put_payload(
-        &mut self,
-        key: ResourceKey,
-        req: DataRequest,
-        obj: DataPayload<SerializeMarker>,
-    ) -> Result<(), DataError> {
-        let mut path_buf = self.root.clone();
-        path_buf.push(&*key.writeable_to_string());
-        path_buf.push(&*req.options.writeable_to_string());
-        log::trace!("Writing: {}", req);
-        self.write_to_path(path_buf, obj.get().deref())?;
-        Ok(())
-    }
-
-    /// This function must be called before the [`FilesystemExporter`] leaves scope.
-    /// It is recommended to flush after exporting each [`ResourceKey`].
-    fn flush(&mut self) -> Result<(), DataError> {
-        if let Some(mut alias_collection) = self.alias_collection.take() {
-            alias_collection.flush()?;
-        }
-        Ok(())
-    }
+    // The presence of the option does not change for the lifetime of the object and
+    // therefore doesn't need to be locked. This way a non-aliasing instance is lock-free.
+    alias_collections: Option<Mutex<BTreeMap<ResourceKeyHash, AliasCollection<Vec<u8>>>>>,
```

Comment:
Suggestion: This looks far more complicated than it's worth. I think we should remove this from FsDataProvider and have files always written out. Deduplication belonds in BlobProvider.